frontend/src/concepts/pipelines/content/createRun/submitUtils.ts

```diff
@@ -129,20 +132,30 @@ export const handleSubmit = (formData: RunFormData, api: PipelineAPIs): Promise<
 const normalizeInputParams = (
   params: RuntimeConfigParameters,
   version: PipelineVersionKFv2 | null,
-): RuntimeConfigParameters => {
-  const inputDefinitionParams = version?.pipeline_spec?.root?.inputDefinitions?.parameters;
-
-  return Object.entries(params).reduce((acc: RuntimeConfigParameters, [paramKey, paramValue]) => {
+): RuntimeConfigParameters =>
+  Object.entries(params).reduce((acc: RuntimeConfigParameters, [paramKey, paramValue]) => {
+    const inputDefinitionParams = getInputDefinitionParams(version);
     const paramType = inputDefinitionParams?.[paramKey].parameterType;
 
-    if (paramType === InputDefinitionParameterType.Boolean) {
-      acc[paramKey] = paramValue === 'true';
-    } else if (paramType === InputDefinitionParameterType.NumberInteger) {
-      acc[paramKey] = paramValue ? parseInt(paramValue.toString()) : 0;
-    } else {
-      acc[paramKey] = paramValue;
+    switch (paramType) {
+      case InputDefParamType.Boolean:
+        acc[paramKey] = paramValue === 'true';
+        break;
+      case InputDefParamType.NumberInteger:
+        acc[paramKey] = paramValue ? parseInt(String(paramValue)) : 0;
+        break;
+      case InputDefParamType.NumberDouble:
+        acc[paramKey] = paramValue ? parseFloat(String(paramValue)) : 0;
+        break;
+      case InputDefParamType.Struct:
+        acc[paramKey] = paramValue ? JSON.parse(String(paramValue)) : {};
+        break;
+      case InputDefParamType.List:
```

Comment:
if you dont fill out the boolean inputs it defaults to false, even when the form shows nothing selected.

Reply:
Yes, that was an intentional fallback I thought made sense.

Reply:
she we just block the form from submission. KF requires all the fields to be filled in, and if im not mistaken i thought we did that too? I need to check. But anyways If we are going to default to false, or 0 in the case of ints and floats, then that is what the default value should be, or just make them required fields.

@xianli123 how should we handle this. should the boolean values start false or should they be required fields?

Reply:
I marked them all as required and removed the fallbacks. They are required, which is why I added the fallbacks to make it easier to submit. But I can see how that might not be desired from a user perspective.  I can also see how fallbacks could be useful still, so I can revert to fallbacks based on design feedback.

Reply:
Right the true/false is where it gets tricky. We can wait until @xianli123 or @yannnz chime in

Reply:
Hi @jpuzz0 @Gkrumbach07 all the parameters should be required. The parameters are defined by the selected pipeline. If the parameters are displayed, that means the pipeline has set these displayed parameters. 
Could you please share a screenshot of this section that you were talking about in case I misunderstood this part? Thanks!

Reply:
Hi @jpuzz0  The pipeline definition in the code determines which parameters appear in the UI form. All the parameters are required by the selected pipeline(I attached a screenshot below). So I think users should not be able to leave the field blank (please correct me if I misunderstood the logic). so if the field is blank, we could show an error status. It may look like the second screenshot. Does it make sense?

<img width="592" alt="Screenshot 2024-03-11 at 22 35 06" src="https://github.com/opendatahub-io/odh-dashboard/assets/53931519/3f80d8b3-4a2a-4e45-bd15-d36591e4163b">

<img width="680" alt="Screenshot 2024-03-11 at 22 55 34" src="https://github.com/opendatahub-io/odh-dashboard/assets/53931519/e7bab68a-17a2-4650-b2cc-8703d0bfdce6">

Reply:
This was addressed in slack already, but I'll also add that if the field is blank, we're not enabling the submit button, like the rest of the form validation functions right now.